src/NAR/Constructs/DivideConstruct.java

```diff
@@ -66,34 +65,36 @@ public void setArgs(String statement) {
                 } else {
                     op2 = Double.parseDouble(components[1]);
                 }
+
+            }
+            if(op2==0){
+                System.out.println("Divide by Zero Error: Right operand was 0.");
```

Comment:
make this check in checkSyntax(you will to check the value of the op2 if it is a variable) and also print it to the editor console